source/math/FloatMatrix.ooc

```diff
@@ -16,58 +16,56 @@
 
 use ooc-base
 import math
-import IntVector2D
-import FloatPoint3D
 
-FloatMatrix : cover {
+FloatMatrix: cover {
 	// x = column
 	// y = row
-	_dimensions: IntVector2D
-	dimensions: IntVector2D { get {
-		result := this _dimensions
-		this free(Owner Receiver)
-		result
-	}}
+	_width: Int
 	width: Int { get {
-		result := this _dimensions x
+		result := this _width
 		this free(Owner Receiver)
 		result
 	}}
+	_height: Int
 	height: Int { get {
-		result := this _dimensions y
+		result := this _height
 		this free(Owner Receiver)
 		result
 	}}
 	isNull: Bool { get { // TODO: Better name?
-		result := this _dimensions empty
+		result := this _width == 0 || this _height == 0
```

Comment:
Replacing this line could potentially give different results:
```ooc
IntVector2D {
	empty ::= !(this x > 0 && this y > 0)
}
```
https://github.com/cogneco/ooc-kean/issues/870